store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -201,25 +194,23 @@ private boolean remove(final String graphId, final Predicate<FederatedAccess> ac
     }
 
     private void removeGraphCaches(final String graphId) {
-        if (CacheServiceLoader.isEnabled()) {
-            try {
-                final GraphSerialisable graphFromCache = federatedStoreCache.getGraphFromCache(graphId);
-                final StoreProperties storeProperties = graphFromCache.getStoreProperties();
-                final ArrayList<String> cacheNames = Lists.newArrayList(
-                        NamedViewCache.getCacheNameFrom(storeProperties.getCacheServiceNamedViewSuffix(graphId)),
-                        NamedOperationCache.getCacheNameFrom(storeProperties.getCacheServiceNamedOperationSuffix(graphId)),
-                        JobTracker.getCacheNameFrom(storeProperties.getCacheServiceJobTrackerSuffix(graphId)));
-                for (final String cacheName : cacheNames) {
-                    final ICache<Object, Object> cache = CacheServiceLoader.getService().getCache(cacheName);
-                    if (nonNull(cache)) {
-                        cache.clear();
-                    } else {
-                        LOGGER.debug(String.format("No cache found graphId:%s with cacheName:%s", graphId, cacheName));
-                    }
+        try {
+            final GraphSerialisable graphFromCache = federatedStoreCache.getGraphFromCache(graphId);
+            final StoreProperties storeProperties = graphFromCache.getStoreProperties();
+            final List<Cache<?, ?>> caches = Lists.newArrayList(
+                    new NamedViewCache(storeProperties.getCacheServiceNamedViewSuffix(graphId)),
+                    new NamedOperationCache(storeProperties.getCacheServiceNamedOperationSuffix(graphId)),
+                    new JobTracker(storeProperties.getCacheServiceJobTrackerSuffix(graphId)));
+            for (final Cache<?, ?> cacheInstance : caches) {
+                final ICache<?, ?> cache = cacheInstance.getCache();
```

Comment:
Doesnt the `Cache` class have a `clearCache` method already? Why does it need to be used to then get the `ICache` first?

Reply:
I think you're right. I'd refactored the existing logic which did this, but it does seem that you could clear the cache directly without these extra steps. I'll look at making this change.

Reply:
Turns out that this wouldn't give much of an advantage as we need to do the null check and `clearCache` doesn't allow for that and would throw and exception. Could do something like this but I don't think it's better:
```
            for (final Cache<?, ?> cacheInstance : caches) {
                if (nonNull(cacheInstance.getCache())) {
                    cacheInstance.clearCache();
                } else {
                    LOGGER.debug(String.format("No cache found graphId:%s with cacheName:%s", graphId, cacheInstance.getCacheName()));
                }
            }
```

Reply:
I don't think the other method would require a null check as if you follow the logic through it will ultimately use the `ICacheService` implementation for the cache instance to do the clearing which already handles the `getCache()` call.

Reply:
I believe the reason for not using `clearCache` in `Cache` is as follows.

For this the `ICacheService` object is fetched and the `clearCache` method for that is called. This method then proceeds to fetch the `ICache` object (using implementation specifc `getCache`) and immediately call its `clear` method. A problem could occur where the `ICacheService` `getCache` method returns a `null` instance of an `ICache` implementation which _might_ then throw an exception when its `clear` method is called.

Fetching the `ICache` itself and checking it is not null avoids this problem. This exception is only an issue because it seems we want to log when no cache is found, rather than throwing an error.

Further changes could be made, e.g. move the null check to `ICacheService`, but I think the current setup is ok for now.